cloudformation/athena.yaml

```diff
@@ -1,99 +1,104 @@
-TransactionDB:
-  Type: AWS::Glue::Database
-  Properties:
-    CatalogId: !Ref AWS::AccountId
-    DatabaseInput:
-      Name: btm-transactions
+AWSTemplateFormatVersion: '2010-09-09'
 
-TransactionTable:
-  Type: AWS::Glue::Table
-  Properties:
-    CatalogId: !Ref AWS::AccountId
-    DatabaseName: !Ref TransactionDB
-    TableInput:
-      Name: btm_transactions
-      TableType: EXTERNAL_TABLE
-      Parameters:
-        has_encrypted_data: "true"
-        projection.enabled: "true"
-        projection.datetime.type: date
-        projection.datetime.range: '2022-01-01,NOW'
-        projection.datetime.format: 'yyyy-MM-dd'
-        projection.datetime.interval: 1
-        projection.datetime.interval.unit: DAYS
-        storage.location.template: !Join [ '', [ 's3://', !Ref StorageBucket, '/btm_transactions/${datetime}/' ] ]
-      PartitionKeys:
-        - { Name: "datetime", Type: date }
-      StorageDescriptor:
-        Columns:
-          - { Name: "event_id", Type: string }
-          - { Name: "event_name", Type: string }
-          - { Name: "client_id", Type: string }
-          - { Name: "timestamp", Type: timestamp }
-          - { Name: "timestamp_formatted", Type: string }
-          - { Name: "component_id", Type: string }
-          - { Name: "user", Type: string }
-          - { Name: "platform", Type: string }
-          - { Name: "restricted", Type: string }
-          - { Name: "extensions", Type: string }
-        Compressed: true
-        InputFormat: "org.apache.hadoop.mapred.TextInputFormat"
-        Location: !Join [ '', [ 's3://', !Ref StorageBucket, '/' ] ]
-        StoredAsSubDirectories: false
-        OutputFormat: "org.apache.hadoop.hive.ql.io.IgnoreKeyTextOutputFormat"
-        SerdeInfo:
-          Parameters: { "ignore.malformed.json": true,"serialiazation.format": 1,"field.delim": "" }
-          SerializationLibrary: "org.openx.data.jsonserde.JsonSerDe"
+Resources:
+  TransactionDB:
+    Type: AWS::Glue::Database
+    Properties:
+      CatalogId: !Ref AWS::AccountId
+      DatabaseInput:
+        Name: !Join [ '-', [!Ref Prefix, !Ref Environment, transactions ] ]
```

Comment:
What is the Prefix? !Ref Prefix means that it will return the value of Prefix right? Where is that declared?

Reply:
Prefix is one of the parameters this stack has. They are defined in `template-source.yaml`. This is to be easily able to rename everything in case we want to split into multiple stacks at some point.